src/editor/CSSInlineEditor.js

```diff
@@ -151,6 +151,34 @@ define(function (require, exports, module) {
         return html;
     }
 
+    function checkProvider(hostEditor,pos){
+
+      // can make htmlToCSSProvider call this later
+
+      // Only provide a CSS editor when cursor is in HTML content
+      if (hostEditor.getLanguageForSelection().getId() !== "html") {
+          return null;
+      }
+
+      // Only provide CSS editor if the selection is within a single line
+      var sel = hostEditor.getSelection();
+      if (sel.start.line !== sel.end.line) {
+          return null;
+      }
+
+      // Always use the selection start for determining selector name. The pos
+      // parameter is usually the selection end.
+      var selectorResult = _getSelectorName(hostEditor, sel.start);
+      if (selectorResult.selectorName === "") {
+          return selectorResult.reason || null;
+      }
+
+      var selectorName = selectorResult.selectorName;
+
+      var result = new $.Deferred();
```

Comment:
We don't need this to be async, you should just be returning a `true/false` value to indicate whether there is a provider.